File: leetcode/0061-rotate-list/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def rotateRight(self, head, k):
        if not head or  not head.next or k==0:
            return head
        t = head
        l = 1
        while t.next:
            t = t.next
            l+=1
        
        k = k % l
        if k ==0:
            return head
        curr = head
        for _ in range(l - k -1):
            curr = curr.next
        
        new_head = curr.next
        t.next = head
        curr.next = None
        return new_head


        

        """
        :type head: Optional[ListNode]
        :type k: int
        :rtype: Optional[ListNode]
        """
        

File: leetcode/0061-rotate-list/solution.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function rotateRight(head: ListNode | null, k: number): ListNode | null {
    if(!head || !head.next || k==0){
        return head
    }

    let curr = head
    let len = 1
    while(curr.next){
        len++
        curr = curr.next
    }
    k = k%len
    if(k===0){
        return head
    }
    let cur = head
    for(let l =0;l<len-k-1;l++){
        cur = cur.next
    }
    let newHead = cur.next
    curr.next = head
    cur.next= null

    return newHead
};
